modules/optimization/src/optimizationreporters/GeneralOptimization.C

```diff
@@ -97,6 +95,12 @@ GeneralOptimization::getNumParams() const
 void
 GeneralOptimization::setICsandBounds()
 {
+  // Check that one and only one set of parameters are given.
+  // Set here because some derived reporters use a different method of
+  // determining numbers of dofs
+  if (!(isParamValid("num_values_name") ^ isParamValid("num_values")))
```

Comment:
What does this if condition do?  is it bool raised to the bool power?

Reply:
`^` is the logical XOR operator.